system-image/default.nix

```diff
@@ -13,7 +15,10 @@ let nixos = pkgs.importFromNixos ""; in
 
       # Application-specific module
       application.module
-    ];
+    ]
+    # TODO: move this to a 'test' overlay in the application?
+    ++ lib.lists.optional isTestBuild
+        (import ../testing/end-to-end/profile.nix { inherit pkgs; });
```

Comment:
Yes, I think it would be better not to have this explicit dependency on test instrumentation here.

Ideally this would have become part of the application by the time it reaches here.

Reply:
Do you mean something like this? https://github.com/yfyf/playos/commit/1ce6a8f82efa5eab9b9f6c3088bd636d9f382314

Unfortunately this leads to `error: infinite recursion encountered` and I can't figure out what is causing it.